redhat_access_insights/connection.py

```diff
@@ -169,6 +153,13 @@ def get_proxies(self, config):
                 conf_proxy = scheme + location
             logger.debug("CONF Proxy: %s", conf_proxy)
             proxies = {"https": conf_proxy}
+
+        if os.environ.get('NO_PROXY') and conf_proxy:
+            logger.debug("Invalid Proxy configuration. "
+                         "You have environment variable NO_PROXY set "
+                         "as well as 'proxy' set in your configuration file. "
+                         "This is invalid. Do not expect normal results.")
```

Comment:
I think we are over-analysing this one. Lets keep it simple:
(1) We need to keep the conf file setting (for the reasons Ian pointed out - can use puppet to configure etc)
(2) Our code uses proxy defined  in config file, if there is one, and ignores HTTPS_PROXY and NO_PROXY configuration in environment variables
(3) If there is no proxy configuration in config file, fall back to using environment variables, if defined

This takes us back to the original patch. Note that message about the NO_PROXY variable being ignored is there for troubleshooting, just in case.